src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -284,62 +284,119 @@ std::string Runnable::generate_output_filepath() {
     return filename.str();
 }
 
+void Runnable::contact_bouncer(Callback<Error> cb) {
+    auto bouncer = options.get("bouncer_base_url",
+            ooni::bouncer::production_bouncer_url());
+    logger->info("Contacting bouncer: %s", bouncer.c_str());
+    ooni::bouncer::post_net_tests(
+        bouncer, test_name, test_version, test_helpers_bouncer_names(),
+        [=](Error error, Var<BouncerReply> reply) {
+            if (error) {
+                cb(error);
+                return;
+            }
+            assert(!!reply);
+            auto maybe_collector = reply->get_collector_alternate("https");
+            if (!maybe_collector) {
+                cb(maybe_collector.as_error());
+                return;
+            }
+            logger->info("Bouncer discovered collector: %s",
+                         maybe_collector->c_str());
+            if (options.find("collector_base_url") == options.end()) {
```

Comment:
According with the assertion **"If we don't find the collector ooniprobe will fail"** this code should be refactored avoiding this check and directly setting the `options["collector_base_url"]` with `*maybe_collector`